File: cognipy/CogniPyLib/Encode/MD5.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;
using System.IO;

namespace Ontorion.Common
{
    internal static class MD5
    {
        /// <summary>
        /// Calculates a MD5 hash from the given string and uses the given
        /// encoding.
        /// </summary>
        /// <param name="Input">Input string</param>
        /// <param name="UseEncoding">Encoding method</param>
        /// <returns>MD5 computed string</returns>
        private static string Calculate(string Input, Encoding UseEncoding)
        {
            System.Security.Cryptography.MD5CryptoServiceProvider CryptoService;
            CryptoService = new System.Security.Cryptography.MD5CryptoServiceProvider();

            byte[] InputBytes = UseEncoding.GetBytes(Input);
            InputBytes = CryptoService.ComputeHash(InputBytes);
            return BitConverter.ToString(InputBytes).Replace("-", "");
        }

        /// <summary>
        /// Calculates a MD5 hash from the given string. 
        /// (By using the default encoding)
        /// </summary>
        /// <param name="Input">Input string</param>
        /// <returns>MD5 computed string</returns>
        public static string Calculate(string Input)
        {
            // That's just a shortcut to the base method
            return Calculate(Input, System.Text.Encoding.UTF8);
        }
    }
}

File: cognipy/cognipylib/cognipy/common/MD5.java
package cognipy.common;

import cognipy.*;

public final class MD5
{
	/** 
	 Calculates a MD5 hash from the given string and uses the given
	 encoding.
	 
	 @param Input Input string
	 @param UseEncoding Encoding method
	 @return MD5 computed string
	*/
	private static String Calculate(String Input, Encoding UseEncoding)
	{
		System.Security.Cryptography.MD5CryptoServiceProvider CryptoService;
		CryptoService = new System.Security.Cryptography.MD5CryptoServiceProvider();

//C# TO JAVA CONVERTER WARNING: Unsigned integer types have no direct equivalent in Java:
//ORIGINAL LINE: byte[] InputBytes = UseEncoding.GetBytes(Input);
		byte[] InputBytes = UseEncoding.GetBytes(Input);
		InputBytes = CryptoService.ComputeHash(InputBytes);
		return BitConverter.toString(InputBytes).replace("-", "");
	}

	/** 
	 Calculates a MD5 hash from the given string. 
	 (By using the default encoding)
	 
	 @param Input Input string
	 @return MD5 computed string
	*/
	public static String Calculate(String Input)
	{
		// That's just a shortcut to the base method
		return Calculate(Input, System.Text.Encoding.UTF8);
	}
}